hazelcast/src/main/java/com/hazelcast/spi/impl/BasicOperationService.java

```diff
@@ -145,11 +144,16 @@
         int concurrencyLevel = reallyMultiCore ? coreSize * CORE_SIZE_FACTOR : CONCURRENCY_LEVEL;
         this.invocations = new ConcurrentHashMap<Long, BasicInvocation>(INITIAL_CAPACITY, LOAD_FACTOR, concurrencyLevel);
         this.operationHandler = new OperationHandler();
-        this.responsePacketHandler = new ResponsePacketHandler();
         this.operationBackupHandler = new OperationBackupHandler();
+
+        ExecutionService executionService = nodeEngine.getExecutionService();
+        ResponsePacketHandler responsePacketHandler = new ResponsePacketHandler();
         this.scheduler = new BasicOperationScheduler(node, executionService, operationHandler, responsePacketHandler);
-        this.asyncExecutor = executionService.register(ExecutionService.ASYNC_EXECUTOR, coreSize,
-                ASYNC_QUEUE_CAPACITY, ExecutorType.CONCRETE);
+        this.asyncExecutor = executionService
+                .register(ExecutionService.ASYNC_EXECUTOR, coreSize, ASYNC_QUEUE_CAPACITY, ExecutorType.CONCRETE);
+
+        this.slowOperationDetector = new SlowOperationDetector(node, this.scheduler);
```

Comment:
Try to inject concrete dependencies you use instead of Node. Now the whole things becomes only testable with a whole cluster up and running.